Dialogue requires Java 11

carterkozak opened the pull request:
==COMMIT_MSG==
 Dialogue requires Java 11
==COMMIT_MSG==

carterkozak committed:
Dialogue requires Java 11

carterkozak committed:
avoid finalizers

svc-changelog committed:
Add generated changelog entries

carterkozak commented on dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientChannels.java:
```diff
@@ -235,21 +199,132 @@ static CloseableClient wrap(
                     .build();
             createMeter.mark();
-            return newInstance;
+            CloseableClient wrapper = new CloseableClientWrapper(newInstance);
+            CleanerSupport.register(wrapper, newInstance::closeApacheClient);
+            return wrapper;
```
refactor exists for these three lines -- we must create a delegate which can be strongly referenced by the cleaner without blocking the sentinel reference (the wrapper) from being garbage collected. When the outer class is garbage collected, our cleaner is able to execute.

jkozlowski replied:
Discussed: I found this kinda hard to reason about, the cleaner stuff always is fun. We discussed instead having private static final InternalCloseableClient and public static final class CloseableClient that takes InternalCloseableClient and delegates. I find that easier to understand, and Carter doesn't so let's just ship this.

jkozlowski commented on build.gradle:
```diff
@@ -60,7 +60,7 @@ subprojects {
     apply plugin: 'org.inferred.processors'
 
-    sourceCompatibility = 1.8
+    sourceCompatibility = 11
```
Can you remove overrides in dialogue-annotations-example and dialogue-annotations-processor?

carterkozak replied:
Also removed the `-source 11` lines from the test javac runner since it's the new default.

jkozlowski commented on dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ResponseLeakDetector.java:
```diff
@@ -71,9 +72,6 @@ Response wrap(Response input, Endpoint endpoint) {
     }
 
     private boolean shouldApplyLeakDetection() {
```
While you're here, can you remove the whole shouldApplyLeakDetection? probability is 1 now, so all this code can probably go away?

carterkozak replied:
good catch!

jkozlowski reviewed (commented):
I think there is 1 action and maybe another action (but can be in a FLUP PR)

jkozlowski reviewed (approved):
approving to unblock, since I'm going offline shortly

carterkozak committed:
simplify ResponseLeakDetector -- always sample

We've sampled at 100% for a while, now the code is easier to read
and doesn't appear configurable.

carterkozak committed:
Remove unnecessary sourceCompatibility overrides

svc-autorelease commented:
Refusing to release a major rev - please do this manually at https://autorelease.bots.palantir.build/palantir/dialogue
